addons/mail/static/src/messaging/component/notification_list/notification_list_tests.js

```diff
@@ -0,0 +1,216 @@
+odoo.define('mail.messaging.component.NotificationListTests', function (require) {
+'use strict';
+
+const NotificationList = require('mail.messaging.component.NotificationList');
+const {
+    afterEach: utilsAfterEach,
+    afterNextRender,
+    beforeEach: utilsBeforeEach,
+    pause,
+    start: utilsStart,
+} = require('mail.messaging.testUtils');
+
+QUnit.module('mail', {}, function () {
+QUnit.module('messaging', {}, function () {
+QUnit.module('component', {}, function () {
+QUnit.module('NotificationList', {
+    beforeEach() {
+        utilsBeforeEach(this);
+
+        /**
+         * @param {Object} param0
+         * @param {string} [param0.filter='all']
+         */
+        this.createNotificationListComponent = async ({ filter = 'all' }) => {
+            NotificationList.env = this.env;
+            this.notificationList = new NotificationList(null, { filter });
+            await this.notificationList.mount(this.widget.el);
```

Comment:
```js
const NotificationListComponent = components.NotificationList;
NotificationListComponent.env = this.env;
this.component = new NotificationListComponent();
```

- adapt with component namespacing
- use `this.component` over `this.notificationList` (will likely refactor most before/afterEach of component tests in the future)